src/mock_builder.ts

```diff
@@ -96,52 +99,56 @@ export class MockBuilder {
    *     Get all properties--public, protected, private--from the object that
    *     will be mocked.
    *
-   * @param any constructorFn
+   * @param T obj
    *     The object that will be mocked.
    *
    * @return string[]
    *     Returns an array of the object's properties.
    */
-  protected getAllProperties(constructorFn: any): string[] {
+  protected getAllProperties(obj: T): string[] {
     let functions: string[] = [];
-    let clone = constructorFn;
+    let clone = obj;
     do {
       functions = functions.concat(Object.getOwnPropertyNames(clone));
-    } while (clone = Object.getPrototypeOf(clone));
+    } while (!!(clone = Object.getPrototypeOf(clone)));
 
-    return functions.sort().filter(function (e: any, i: number, arr: any[]) {
-      if (
-        e != arr[i + 1] && typeof constructorFn[e] != "function"
-      ) {
-        return true;
-      }
-    });
+    return functions.sort().filter(
+      function (e: string, i: number, arr: unknown[]) {
+        if (
+          e != arr[i + 1] && typeof obj[e as keyof T] != "function"
+        ) {
+          return true;
+        }
+      },
+    );
   }
 
   /**
    * @description
    *     Get all functions--public, protected, private--from the object that
    *     will be mocked.
    *
-   * @param any constructorFn
+   * @param T obj
    *     The object that will be mocked.
    *
    * @return string[]
    *     Returns an array of the object's functions.
    */
-  protected getAllFunctions(constructorFn: any): string[] {
+  protected getAllFunctions(obj: T): string[] {
     let functions: string[] = [];
-    let clone = constructorFn;
+    let clone = obj;
     do {
       functions = functions.concat(Object.getOwnPropertyNames(clone));
-    } while (clone = Object.getPrototypeOf(clone));
+    } while (!!(clone = Object.getPrototypeOf(clone)));
 
-    return functions.sort().filter(function (e: any, i: number, arr: any[]) {
-      if (
-        e != arr[i + 1] && typeof constructorFn[e] == "function"
-      ) {
-        return true;
-      }
-    });
+    return functions.sort().filter(
+      function (e: string, i: number, arr: unknown[]) {
```

Comment:
maybe a lambda would be prettier

Reply:
Yeah feel free to use a lambda if you want, as long as it won't break anything inside :P

Reply:
agreed

Reply:
will leave as a function for rn